### ISSUE TITLE
linter: false positive `promise/prefer-await-to-then` flags Zod `.catch()` as promise chain

### ISSUE THREAD

**Issue by fellipeutaka** [oxc-project/oxc#22161]
### What version of Oxlint are you using?

```
oxlint 1.63.0
oxlint-tsgolint 0.22.1
```

### What command did you run?

`oxlint --plugin promise`

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "plugins": ["promise"],
  "rules": {
    "promise/prefer-await-to-then": "warn"
  }
}
```

### What happened?

The rule `promise/prefer-await-to-then` incorrectly flags `.catch()` calls on non-Promise objects. Zod's `.catch()` method provides a fallback value for schema parsing failures — it has nothing to do with promises — but the rule treats any `.catch()` call as a promise chain.

**Code that triggers the false positive:**

```ts
import { z } from "zod";

const schema = z
  .preprocess(safeParseJsonArray, z.array(z.object({ id: z.string() })))
  .catch([])    // ← flagged here
  .default([]);
```

**Diagnostic reported:**

```
Prefer await to then()/catch()/finally()
help: Use `await` with `try`/`catch` instead of promise chaining for more readable and maintainable async code. (oxc eslint-plugin-promise(prefer-await-to-then))
```

**Expected behavior:**

No diagnostic. The receiver of `.catch([])` is a `ZodEffects` schema object, not a `Promise` or `PromiseLike`. The rule should only flag `.catch()` / `.then()` / `.finally()` when called on an actual Promise.

**Impact:**

Any library that uses `.catch()` as part of its own fluent API (Zod, RxJS, custom result types, etc.) triggers this false positive, forcing `// oxlint-disable-next-line` suppressions on legitimate non-promise code.

**Possible fix:**

Use TypeScript type information to check whether the callee is a `Promise` / `PromiseLike` before reporting. If type info is unavailable, consider restricting the rule to `.then()` calls only, since `.then()` is the canonical thenable signal, unlike `.catch()` which is widely used in non-promise fluent APIs.

**Comment by camc314** [comment]
Hi,

thanks for filing this issue.

I'm closing this as not planned - this rule does not use type information and this report will match the upstream rule:

https://github.com/eslint-community/eslint-plugin-promise/blob/main/docs/rules/prefer-await-to-then.md

Unfortunately this can result in false positives.

Thanks
